### PR TITLE
A bit of everything

### PR DESCRIPTION
**Pull request by haskellcamargo**
This pull-request applies several changes to Quack:

- Remove unused `&` references from objects
- Replace `ScopeError` by `TypeError` on `getType` functions
- Cloning the type when returning on `getType`
- HashMaps now start with the `#` symbol (instead of `$`)
- Objects and class constructors now start with `%` symbol, like in Elixir
- Semicolons have been replaced by commas
- `->` for pairs have been replaced by `:`
- Using `~&` instead of `^` for bitmask because the compiler fields use base 2
- Throw error when an object receive duplicated properties
- Fix bug where `1..10` was returning `number` instead of `list.of(number)`
- Add `application/x-quack` mimetype, icons and installation script:

image

### CODE REVIEW

File: src/ast/expr/OperatorExpr.php [hunk 1]
```diff
@@ -71,11 +72,19 @@ public function injectScope(&$parent_scope)
 
         if (':-' === $this->operator) {
             if ($this->left instanceof NameExpr) {
-                // When it is an attribution by name, ensure the variable is mutable
                 $symbol = $parent_scope->lookup($this->left->name);
 
-                // when symbol is not mutable
-                if ($symbol ^ Kind::K_MUTABLE) {
+                var_dump($symbol, $this->left->name);
```
Comment (luizperes) [inline src/ast/expr/OperatorExpr.php]: I believe you forgot a `var_dump` here

File: unix/src/configure-mimetype.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+copy_icon '16x16'
+copy_icon '32x32'
+copy_icon '48x48'
+copy_icon '64x64'
+copy_icon '128x128'
+
+# Reload configurations
+update-mime-database /usr/share/mime
+update-icon-caches /usr/share/icons/*
```
Comment (luizperes) [inline unix/src/configure-mimetype.sh]: Missing new line at the end of the file. This is not the only one, please check the others :)

**Review by luizperes — CHANGES_REQUESTED** [review]

**Comment by luizperes** [comment]
Yay! LGTM !

**Review by luizperes — APPROVED** [review]

**Comment by sergiors** [comment]
👏👏👏👏👏👏

**Pull request merged**